### ISSUE TITLE
Spellcheck in Constellation

### ISSUE THREAD

**Issue by antares1470** [constellation-app/constellation#870]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Running the latest version of Constellation
	* Attached the ***Support Package*** via `Help` > `Support Package`
    * Checked the FAQs: https://github.com/constellation-app/constellation/wiki/FAQ
    * Checked that your issue isn't already filed: https://github.com/constellation-app/constellation/issues
    * Checked that there is not already a module that provides the described functionality: https://github.com/constellation-app/constellation/wiki/Catalogue-of-Repositories

### Description

For text fields in Data Access View plugins and elsewhere, spell check underlining misspelled words would help prevent typos in what is being submitted.

**Comment by Auriga2** [comment]
Here is the summary of a few options we can use. I am inclined to use `Languagetool` as it has active updates.
[Add spelling check.docx](https://github.com/constellation-app/constellation/files/8337532/Add.spelling.check.docx)

**Comment by Auriga2** [comment]
Based on below comparison we chose JLanguagetool as it has ongoing releases, good documentation and support.
![image](https://user-images.githubusercontent.com/64826928/162897410-f0064512-abdc-42ab-8553-57795527469b.png)

But currently having issues getting it to work when the library is downloaded with ivy. It works in a new Maven project. 

**Performance** 
Regarding the slowness, it was only observed when debugging. When running it was relatively faster. When dealing with multiple threads,

> Create one instance of JLanguageTool per thread, but create the language only once (e.g. new BritishEnglish()) and use that for all instances of JLanguageTool

Also consider [suggestions here](https://forum.languagetool.org/t/why-is-jlanguagetool-not-thread-safe/2525/4): 

> You’ll see that only the very first initializing is slow. After that, it will be fast, no matter from how many threads it’s called. That’s what we do on languagetool.org and it works fine to serve many hundred thousand requests per day from a single machine.

For better performance we can also use [MultiThreadedJLanguageTool](https://languagetool.org/development/api/org/languagetool/MultiThreadedJLanguageTool.html) 


**Ignore Words**
`Ignore` should just leave the current text the user has typed while an `Add to Dictionary` option can be added to permanently ignore it, and consider it as a valid word. In JLanguagetool [`addIgnoreTokens` or `acceptPhrases`](https://dev.languagetool.org/java-api#spell-checking) can be used for that. 
 Further reading suggests that will only work within the same instance of JLanguageTool though. That means it'd not apply in other threads.

**Comment by arcturus2** [comment]
![spelling error](https://user-images.githubusercontent.com/39325530/163969466-12649695-bb53-4334-b4de-0d2bd3cde95f.gif)

My view is that to start with knowing that a word is misspelled is still useful and is a good enough MVP before we look at adding suggestions or adding it to a dictionary. The above example is NetBeans's built in text editor which shows that a word is spelled incorrectly and stops there. It is still useful to know that even if I don't get a suggestion.

**Comment by Auriga2** [comment]
@arcturus2  We can simply add the suggestions as well, ideally I was planning to pop up the suggestions on the left click of a misspelt word, similar to the https://languagetool.org/. That way we have the possibility to leave the right click Context Menu options, hopefully.

**Comment by HolandCrick** [comment]
@Auriga2 - Thanks for enhancing the Spelling Check in Data Access View. It's working nicely in Data Access View only, though need to add this enhancement to other area like Notes View or wherever users might find it useful.

Spelling Check is not working in Notes View:
![image](https://github.com/constellation-app/constellation/assets/157770877/e9d99325-5552-43bc-9f5f-25b33eed1eba)

Created New Ticket #2050

**Comment by antares1470** [comment]
As it turns out, there were a number of breaking issues that were resulting from the indriya dependency which spellcheck added a version of. These have included:
- Schema View throwing errors when loading (#2054)
- Export to GeoPackage and Export to ShapeFile throwing errors

Consequently, it seems this feature wasn't as ready as we thought it was, and so we're re-opening this ticket

**Comment by HolandCrick** [comment]
Spell checking works in Data access view; this is more like a pilot, only added in DAV for now. We will enable it in other areas as required later. **### Ticket # 2050 is for Spelling Check in Notes View.** 

![Image](https://github.com/user-attachments/assets/75a42c6e-19d7-4cd0-8537-b1d3dec7969e)
